### ISSUE TITLE
Response models not generated for paths that contains refs for nested schemas

### ISSUE THREAD

**Issue by pvicol** [koxudaxi/datamodel-code-generator#1084]
**Describe the bug**
It has been observed that when attempting to generate responses for paths that reference nested schemas, the models are not returned.

**To Reproduce**

Example schema:
```yaml

openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
  license:
    name: MIT
  description: |
    This description is for testing
    multi-line
    description

servers:
  - url: http://petstore.swagger.io/v1
security:
  - BearerAuth: []
paths:
  /pets:
    get:
      summary: List all pets
      operationId: listPets
      parameters:
        - name: limit
          in: query
          description: How many items to return at one time (max 100)
          required: false
          schema:
            default: 0
            type: integer
            format: int32
        - name: HomeAddress
          in: query
          required: false
          schema:
            default: 'Unknown'
            type: string
        - name: kind
          in: query
          required: false
          schema:
            default: dog
            type: string
      responses:
        '200':
          description: A paged array of pets
          headers:
            x-next:
              description: A link to the next page of responses
              schema:
                type: string
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: "#/components/schemas/Pet"
        default:
          description: unexpected error
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Error"
components:
  schemas:
    Pet:
      type: object
      required:
        - id
        - name
      properties:
        id:
          type: integer
          format: int64
        name:
          type: string
        tag:
          type: string
    Error:
      required:
        - code
        - message
      properties:
        code:
          type: integer
          format: int32
        message:
          type: string

```

Above schema generates this code:
```python
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class Pet(BaseModel):
    id: int
    name: str
    tag: Optional[str] = None


class Error(BaseModel):
    code: int
    message: str
```

Used commandline:
```
$ datamodel-codegen --input query_parameters.yaml --output model.py --openapi-scopes schemas paths
```

**Expected behavior**
The expected behavior is to create a model:

```python
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class Pet(BaseModel):
    id: int
    name: str
    tag: Optional[str] = None


class Error(BaseModel):
    code: int
    message: str


class PetsGetResponse(BaseModel):
    id: int
    name: str
    tag: Optional[str] = None

```

The schema that does work to generated the expected models:
```yaml
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
  license:
    name: MIT
  description: |
    This description is for testing
    multi-line
    description

servers:
  - url: http://petstore.swagger.io/v1
security:
  - BearerAuth: []
paths:
  /pets:
    get:
      summary: List all pets
      operationId: listPets
      parameters:
        - name: limit
          in: query
          description: How many items to return at one time (max 100)
          required: false
          schema:
            default: 0
            type: integer
            format: int32
        - name: HomeAddress
          in: query
          required: false
          schema:
            default: 'Unknown'
            type: string
        - name: kind
          in: query
          required: false
          schema:
            default: dog
            type: string
      responses:
        '200':
          description: A paged array of pets
          headers:
            x-next:
              description: A link to the next page of responses
              schema:
                type: string
          content:
            application/json:
              schema:
                type: array
                items:
                  required:
                    - id
                    - name
                  properties:
                    id:
                      type: integer
                      format: int64
                    name:
                      type: string
                    tag:
                      type: string
        default:
          description: unexpected error
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Error"
components:
  parameters:
    MyParam:
      name: foo
      in: query
      schema:
        type: string
  schemas:
    Pet:
      required:
        - id
        - name
      properties:
        id:
          type: integer
          format: int64
        name:
          type: string
        tag:
          type: string
    Error:
      required:
        - code
        - message
      properties:
        code:
          type: integer
          format: int32
        message:
          type: string
```

**Version:**
 - OS: macOS Ventura
 - Python version: 3.9
 - datamodel-code-generator version: 0.17.0

**Additional context**
This can also be noticed for current unit tests. There are quite a few OpenAPI spec files have many paths defined, however, the expected outputs do not contains those models.
Examples:
* [body and parameters OpenAPI Spec File](https://github.com/koxudaxi/datamodel-code-generator/blob/master/tests/data/openapi/body_and_parameters.yaml) does not have the generated responses for the following:
  * [openapi_parser_responses_with_tag](https://github.com/koxudaxi/datamodel-code-generator/tree/master/tests/data/expected/parser/openapi/openapi_parser_responses_with_tag) - Scopes in test: Paths, Tags, Schemas
  * https://github.com/koxudaxi/datamodel-code-generator/tree/master/tests/data/expected/parser/openapi/openapi_parser_responses_without_content - Scopes in test: Paths

**Comment by zach-hamm** [comment]
I have the same issue @zupo. What I did to get around it for now is use the swagger-cli to bundle the files together. From there, it works as expected. Definitely a hack/workaround, but it works until the issue can be resolved.

**Comment by koxudaxi** [comment]
@zupo 
@zach-hamm 
I'm sorry I missed your comment.
I confirmed the problem.
I will fix it tonight.

**Comment by koxudaxi** [comment]
I have released the PR as 0.19.0.
The version generates the correct array response instead of a non-array model.
